src/Container/PdoSnapshotStoreFactory.php

```diff
@@ -61,11 +63,13 @@ public function __invoke(ContainerInterface $container): PdoSnapshotStore
         $config = $this->options($config, $this->configId);
 
         $connection = $container->get($config['connection_service']);
+        $serializer = $container->has($config['serialize']) ? $container->get($config['serialize']) : new CallbackSerializer(null, null);
```

Comment:
I don't get that. If `'serializer' => new CallbackSerializer()` is in the defaultOptions how would one override (get it from the container) it without some sort of switch?

```
$serializer = $config['serializer'] instanceof Serializer ? $config['serializer'] : $container->get($config['serialize']);
```